ddpui/core/chart_service.py

```
@@ -0,0 +1,1219 @@
+                    config=config,
+                    is_public=is_public,
+                )
+
+                logger.info(f"Successfully created chart {chart.id} '{title}'")
+                return chart
+
+        except ChartException:
+            raise
+        except ValidationError as e:
+            # Convert Django validation errors to ChartException
+            raise ChartException(error_code=ChartErrorCode.VALIDATION_FAILED, message=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)